Sources/HaCWebsiteLib/Assets.swift

```diff
@@ -0,0 +1,92 @@
+import Kitura
+import SwiftyJSON
+import Foundation
+
+/**
+  Utilities for accessing public assets for the website
+ */
+enum Assets {
+  /**
+    Location of manifest mapping logical file names to real locations (open it for an example)
+   */
+  static private let manifestPath = "./static/dist/manifest.json"
+
+  /**
+    Middleware serving all of our public assets. Where this is mounted on must be kept in
+    sync with `config.urlBase`
+   */
+  static public let fileServingMiddleware = StaticFileServer(path: "./static/dist")
+
+  static private var resolver: AssetResolver? = nil
+  static private var config: AssetsConfig? = nil
+
+  private static func resolveAssetPath(_ assetPath: String) -> String {
+    if (self.resolver == nil) || (!Config.isProduction)  {
+      self.resolver = AssetResolver(manifestPath: manifestPath)
+    }
+
+    if let resolver = self.resolver {
+      return resolver.publicPath(assetPath)
+    }
+    
+    fatalError("Could not create AssetResolver")
+  }
+
+  /**
+    Initialize with configuration. This must be done exactly once before using the public API
+   */
+  public static func initialize(config: AssetsConfig) {
```

Comment:
Why not make Assets a struct with an initialiser?

Reply:
2 reasons:

1. I feel like having a struct with both a static and instance API is confusing. Especially in this case when most of the state is static
2. This way the `urlBase` is kept private - only those configuring the assets can access it

There are ways to get around both of these issues while making it a struct, but this seemed pretty natural.

Reply:
Seems like we're jumping through a lot of hoops with optionals and force unwrapping here for no gain. Why not today this up by moving `resolver`, `config`, and `publicPath` to the instance level. This will also mean we can give Assets a failable initialiser that fails if AssetResolver initialiser fails so we don't have to hold on to an optional resolver either.

Reply:
`resolver` has to be kept separate from `config` as in development is re-created for every request. This is so the manifest can be re-read when the files we are loading changes.

This isn't necessary in production as the file is guaranteed to be the same for the lifetime of an instance, so it's disabled there to stop IO from clogging up our requests.

Reply:
Having an instance version of `Assets` won't stop us from force-unwrapping optionals won't it? It still has to be configured globally by some external thing

Reply:
I see what you're saying about a 'static API'. The issue is that we're using Asset.publicPath(...) in our viewModels right? So if we wanted to use an instance instead we'd need to hold a global instance somewhere. Am thinking of refactoring the routing stuff soon anyway so maybe we can revisit this.

Reply:
Yeah that's precisely the problem.